dvc/command/completion.py

```diff
@@ -0,0 +1,68 @@
+import argparse
+import logging
+
+import shtab
+
+from dvc.command.base import CmdBaseNoRepo, append_doc_link
+
+logger = logging.getLogger(__name__)
+CHOICE_FUNCTIONS = {
+    "bash": {"DVCFile": "_dvc_compgen_DVCFiles"},
+    "zsh": {"DVCFile": "_files -g '(*?.dvc|Dvcfile|dvc.yaml)'"},
+}
+PREAMBLE = {
+    "bash": """
+# $1=COMP_WORDS[1]
+_dvc_compgen_DVCFiles() {
+  compgen -d -S '/' -- $1  # recurse into subdirs
+  compgen -f -X '!*?.dvc' -- $1
+  compgen -f -X '!*Dvcfile' -- $1
+  compgen -f -X '!*dvc.yaml' -- $1
+}
+""",
+    "zsh": "",
+}
+
+
+class Optional(shtab.Optional):
+    DVC_FILE = [shtab.Choice("DVCFile", required=False)]
+
+
+class Required(shtab.Required):
+    DVC_FILE = [shtab.Choice("DVCFile", required=True)]
```

Comment:
```suggestion
class Choices(shtab.Optional):
    DVC_FILE = [shtab.Choice("DVCFile", required=False)]
```

@efiop if we don't want to bother specifying `Required|Optional` in the codebase how about this?

Reply:
@casperdcl specifying that seems useful, I wouldn't remove it.